File: 0024/0024.py
from typing import *

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(next=head)
        currentNode, prevNode = dummy.next, dummy
        while(currentNode and (postNode := currentNode.next)):
            prevNode.next = currentNode.next
            currentNode.next = postNode.next
            postNode.next = currentNode
            prevNode = prevNode.next.next
            currentNode = currentNode.next

        return dummy.next
        

File: 0024/0024.c
#include <stdio.h>

struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* swapPairs(struct ListNode* head){
    struct ListNode* dummy = (struct ListNode* )malloc(sizeof(struct ListNode));
    dummy->next = head;
    struct ListNode* currentNode = dummy->next;
    struct ListNode* prevNode = dummy;
    struct ListNode* postNode = currentNode && currentNode->next ? currentNode->next : NULL;

    while(postNode){
        prevNode->next = postNode;
        currentNode->next = postNode->next;
        postNode->next = currentNode;
        prevNode = prevNode->next->next;
        currentNode = prevNode->next;
        postNode = currentNode && currentNode->next ? currentNode->next : NULL;
    }

    return dummy->next;
}
